### ISSUE TITLE
<CheckToggle/> - implement missing component

### ISSUE THREAD

**Issue by shlomitc** [wix/wix-style-react#3853]
# ✨ Feature Request

### 🔦 Scope

`<CheckToggle/>`

see specs
https://zpl.io/adzrnvL

### 💁 Explanation
![image](https://user-images.githubusercontent.com/6093192/62044233-1187af00-b20b-11e9-9559-07b094906459.png)

**Comment by argshook** [comment]
does it really have to be new component? that looks like `<Checkbox/>` with `rounded` or `theme="rounded"` or similar

**Comment by shlomitc** [comment]
@wuwa - WDYT? does it make sense for it to be a variation of `Checkbox`/`ToggleSwitch`?

**Comment by wuwa** [comment]
yes could be checkbox extension - however it comes with a couple more new props: size and color. it's a bit closer semantically to ToggleSwitch (that has size+color btw), but very far visually. worried mostly about abuse - if it can be round or square for whatever checkbox. so not sure, any suggestions?

**Comment by shlomitc** [comment]
From the API perspective, our `<Checkbox/>` expose the `indeterminate` state, which is not necessary for this component (or is it?)
`<ToggleSwitch/>` might be more likely to have the same API, the css is not an issue here in my opinion, but the name of it will be misleading since it's no longer a "switch".

This reminds me a bit the discussion we had about `Button`. we used to have one button to rule them all, but then we saw it tries to do too much, so we split it semantically to different components (`Button`, `IconButton`, `TextButton`, `CloseButton`) and they all sit on the same `ButtonNext` component.

I think this is also the case here, we have several semantic items that have similar capabilities internally, but it's ok to treat them differently and have the flexibility to change specific things later.
